src/main/java/org/wildfly/security/http/impl/NonceManager.java

```diff
@@ -94,7 +94,7 @@ String generateNonce(byte[] salt) {
 
             ByteBuffer byteBuffer = ByteBuffer.allocate(PREFIX_LENGTH + messageDigest.getDigestLength());
             byteBuffer.putInt(nonceCounter.incrementAndGet());
-            byteBuffer.putLong(System.currentTimeMillis());
+            byteBuffer.putLong(System.nanoTime());
```

Comment:
I guess what we really want to avoid is a situation where sudden system clock adjustments could cause erratic behavior.  But maybe there's no other option, if this relies on synchronized clocks across JVMs.